scripts/rabbitmq-integration-test-2015.sh

```diff
@@ -0,0 +1,283 @@
+#
+# bring up components, submit test crash, ensure that it shows up in
+# reports tables.
+#
+# This uses the same setup as http://socorro.readthedocs.org/en/latest/installation.html
+
+export PATH=$PATH:${VIRTUAL_ENV:-"socorro-virtualenv"}/bin:$PWD/scripts
+
+source scripts/defaults
+
+if [ "$#" != "1" ] || [ "$1" != "--destroy" ]
+then
+  echo "WARNING - this script will destroy the local socorro install."
+  echo "The default database and config files will be overwritten."
+  echo "You must pass the --destroy flag to continue."
+  exit 1
+fi
+
+function cleanup_rabbitmq() {
+  echo -n "INFO: Purging rabbitmq queue 'socorro.normal'..."
+  python scripts/test_rabbitmq.py --test_rabbitmq.purge='socorro.normal' --test_rabbitmq.rabbitmq_host=$rmq_host --test_rabbitmq.rabbitmq_user=$rmq_user --test_rabbitmq.rabbitmq_password=$rmq_password --test_rabbitmq.rabbitmq_vhost=$rmq_virtual_host > /dev/null 2>&1
+  echo " Done."
+  echo -n "INFO: Purging rabbitmq queue 'socorro.priority'..."
+  python scripts/test_rabbitmq.py --test_rabbitmq.purge='socorro.priority' --test_rabbitmq.rabbitmq_host=$rmq_host --test_rabbitmq.rabbitmq_user=$rmq_user --test_rabbitmq.rabbitmq_password=$rmq_password --test_rabbitmq.rabbitmq_vhost=$rmq_virtual_host > /dev/null 2>&1
+  echo " Done."
+}
+
+function cleanup() {
+  cleanup_rabbitmq
+
+  echo "INFO: cleaning up crash storage directories"
+  rm -rf ./primaryCrashStore/ ./processedCrashStore/
+  rm -rf ./crashes/
+  rm -rf ./submissions
+
+  echo "INFO: Terminating background jobs"
+
+  for p in collector processor middleware
+  do
+    # destroy any running processes started by this shell
+    kill `jobs -p`
+    # destroy anything trying to write to the log files too
+    fuser -k ${p}.log > /dev/null 2>&1
+  done
+
+  return 0
+}
+
+trap 'cleanup' INT
+
+function fatal() {
+  exit_code=$1
+  message=$2
+
+  echo "ERROR: $message"
+  cat setupdb.log
+
+  cleanup
+
+  exit $exit_code
+}
+
+echo -n "INFO: setting up environment..."
+source ${VIRTUAL_ENV:-"socorro-virtualenv"}/bin/activate >> setup.log 2>&1
+if [ $? != 0 ]
+then
+  fatal 1 "could activate virtualenv"
+fi
+export PYTHONPATH=.
+echo " Done."
+
+echo -n "INFO: setting up database..."
+socorro setupdb --dropdb --force --createdb > setupdb.log 2>&1
+if [ $? != 0 ]
+then
+  fatal 1 "setupdb_app.py failed, check setupdb.log"
+  cat setupdb.log
+fi
+echo " Done."
+popd >> setupdb.log 2>&1
+
+# ensure rabbitmq is really empty and no previous failure left garbage
+cleanup_rabbitmq
+
+echo -n "INFO: setting up 'weekly-reports-partitions' via crontabber..."
+python socorro/cron/crontabber_app.py \
+    --job=weekly-reports-partitions \
+    --force \
+    >> setupdb.log 2>&1
+if [ $? != 0 ]
+then
+  fatal 1 "crontabber weekly-reports-partitions failed, check setupdb.log"
+fi
+echo " Done."
+
+echo -n "INFO: configuring backend jobs..."
+for p in collector processor middleware
+do
+  cp config/${p}.ini-dist config/${p}.ini
```

Comment:
Instead of being destructive of the user's config, why not copy it to a different, temporary folder that you would then use in the services?